web-services/security/src/test/java/datawave/security/authorization/remote/RemoteUserOperationsImplHttpTest.java

```diff
@@ -142,15 +216,33 @@ public void after() {
         }
     }
 
+    private void setListEffectiveAuthResponse(SubjectIssuerDNPair userDN, Authorizations auths) {
+        listEffectiveAuthResponse = new DefaultAuthorizationsList();
+        listEffectiveAuthResponse.setUserAuths(userDN.subjectDN(), userDN.issuerDN(), Arrays.asList(auths.toString().split(",")));
+        listEffectiveAuthResponse.addAuths(userDN.subjectDN(), userDN.issuerDN(), Arrays.asList(auths.toString().split(",")));
+    }
+
     @Test
     public void testRemoteUserOperations() throws Exception {
-        DatawavePrincipal principal = new DatawavePrincipal(commonName);
 
-        AuthorizationsListBase auths = remote.listEffectiveAuthorizations(principal);
-        assertEquals(2, auths.getAllAuths().size());
+        AuthorizationsListBase returnedAuths = remote.listEffectiveAuthorizations(principal);
+        assertEquals(2, returnedAuths.getAllAuths().size());
 
         GenericResponse flush = remote.flushCachedCredentials(principal);
         assertEquals("test flush result", flush.getResult());
+
+        ProxiedUserDetails returnedUser = remote.getRemoteUser(principal);
+
+        // ensure that we get the cached user details
+        ProxiedUserDetails dupeReturnedUser = remote.getRemoteUser(principal);
+        assertEquals(returnedUser, dupeReturnedUser);
+
+        // setup the list effective auth response for the other user
+        setListEffectiveAuthResponse(otherUserDN, auths);
+
+        // ensure that we get the other user details, not the cached user details
+        ProxiedUserDetails newReturnedUser = remote.getRemoteUser(otherPrincipal);
+        assertNotEquals(returnedUser, newReturnedUser);
```

Comment:
add an assert for the expected auths as well as that they are different from the cached auths?

Reply:
When you get a cache hit you literally get the exact same object (down to the reference) so i think this is sufficient..